fwb/app/api/discounts/route.ts

```diff
@@ -46,85 +37,212 @@ export async function POST(request: NextRequest, response: NextResponse) {
+    }
   } catch (error) {
-    return NextResponse.json({ error: 'Internal Server Error' }, { status: 500 });
+    return NextResponse.json(
+      { error: "Internal Server Error" },
+      { status: 500 }
+    );
   }
 }
-  
 
 // Delete a discount
-export async function DELETE(
-  request: NextRequest,
-  response: NextResponse
-) {
+export async function DELETE(request: NextRequest, response: NextResponse) {
   try {
     const { userId, getToken, orgRole } = auth();
     const user = await currentUser();
 
     if (userId && user) {
-      const discount_id = request.nextUrl.searchParams.get('discount_id');
+      const discount_id = request.nextUrl.searchParams.get("discount_id");
 
       // Create a Supabase client with the current user's access token
-      const token = await getToken({ template: "supabase" });
+      const token = request.headers.get("supabase_jwt");
       if (!token) {
-        return NextResponse.json({ error: 'Unauthorized' }, { status: 401 })
+        return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
       }
       const supabase = await supabaseClient(token);
 
       const { error } = await supabase
-        .from('discounts')
+        .from("discounts")
         .delete()
-        .eq('id', discount_id)
+        .eq("id", discount_id);
 
       if (error) {
-        return NextResponse.json({ error: 'Failed to delete discount' }, { status: 500 })
+        return NextResponse.json(
+          { error: "Failed to delete discount" },
+          { status: 500 }
+        );
       } else {
-        return NextResponse.json({ success: true }, { status: 200 })
+        return NextResponse.json({ success: true, deleted: discount_id }, { status: 200 });
       }
     } else {
-      return NextResponse.json({ error: 'Unauthorized' }, { status: 401 })
+      return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
     }
   } catch (error) {
-    return NextResponse.json({ error: 'Internal Server Error' }, { status: 500 })
+    return NextResponse.json(
+      { error: "Internal Server Error" },
+      { status: 500 }
+    );
   }
 }
 
 // Update a discount
-export async function PATCH(
-  request: NextRequest,
-  response: NextResponse
-) {
-  return NextResponse.json({ error: 'Not Implemented' }, { status: 500 })
-}
+export async function PATCH(request: NextRequest, response: NextResponse) {
+  try {
+    const { userId, getToken } = auth();
+    const user = await currentUser();
+    if (userId && user) {
+      // Create a Supabase client with the current user's access token
+      const token = request.headers.get("supabase_jwt");
+      if (!token) {
+        console.log("no token")
+        return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+      }
+      const supabase = await supabaseClient(token);
 
+      const formData = await request.formData();
+      const updatedDiscount: any = {};
+
+      const discount_id = formData.get("discount_id");
```

Comment:
Do we need to check each field like this? Is there as way to just update each entry as we get them?